### PR TITLE
Fix free items and guard

### LINKED ISSUE #4111: Builder is building the Farm 2 blocks at a time
### Minecolonies version
Version: minecolonies-0.10.493-RELEASE-universal

### Expected behavior
Like previous buildings, the builder should take all the materials, walk to the area designated, and start building, until it gets a problem, like lack of supplies, or lack of tools.

### Actual behaviour
The builder goes to the builders hut, hangs out for a second, then runs to the build area. He places 2 blocks down, and then goes back to the builders hut. When he was building the Mine, he took all the supplies, ran to the area, and kept building until he got an error. I started the build over multiple times, seeing if that would fix it, but not luck. Here is a few gifs

https://imgur.com/a/vboU2Zl

### Steps to reproduce the problem
Not entirely sure. I have multiple builds pending, with no build order yet. I also destroyed and replaced the builders hut box, because it was facing the wrong way. I let the building "rebuild," and the builder just replaced the torches, and the build was complete.

### Edit
So it turns out the error was only when it came to the floor using the grass path blocks. Once it actually started building the building, he worked as intended

### LINKED ISSUE #4159: Sometimes Crafters keep on making resources until request is fullfilled
### Minecolonies version

Version: minecolonies-0.10.600-ALPHA-universal

### Expected behavior

- Crafters only crafting the requested amount of resources.
### Actual behaviour

- Sometimes when Crafters finish the requested resources they don't successfully send a message to the request system that they finished it to let the dman know they can pick it up. Instead they just keep on secretly requesting resources to try to fullfill the request. This way the dman is being occupied all the time by bringing the crafter the resources he needs (if it is a small amount of resources).

### Steps to reproduce the problem

I tried to reproduce it but I couldnt. It just appears to be super random.

### LINKED ISSUE #4171: Free items when cancelling builder
Version:  minecolonies-0.10.602-ALPHA-universal.jar 

Sometimes when I cancel requests from the builder through the handheld clipboard, instead of the item coming back up as a request later it just gives the builder the item for free. He goes off and places it.

### Steps to reproduce the problem
1. Have a builder building something and catch on an item he's missing
2. Use the handheld clipboard to view his current request
3. Sometimes, I'm not sure why, he'll start showing multiple requests open - not just the next 1 thing he needs but the next 3 things -- not sure if this is relevant but I saw this the two times I ran into the dup bug
4. Cancel his requests using the handheld clipboard. I cancelled them trying to get him to recognize some other item I had put into his hut inventory but he hadn't picked up on yet.
5. He gets the item for free from somewhere. This happened for a lime bed and a fern (two separate incidents hours apart). In both cases he only needed 1 of those items. I am pretty sure I didn't have the fern anywhere in my warehouse system, and very certain I had never crafted a lime bed yet on that play through.
6. Profit. Free lime bed.

Note, I cancel requests through the clipboard all the time without this happening. It isn't every time.

### LINKED ISSUE #4187: Field won't accept pumpkin seeds
### Minecolonies version

Version: minecolonies-0.10.493-RELEASE-universal

### Expected behavior

- Insert pumpkin seeds into field

### Actual behaviour

- Cannot insert pumpkin seeds into field.

### Steps to reproduce the problem

1. Build farm (medieval oak) and insert wheat seeds into the first field.
2. Upgrade farm to level 2 and create 2nd field. 
3.  Try putting pumpkin seeds in field. 
4. ... Profit

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #4171
Closes #4111
Closes #4187
Closes #4159

# Changes proposed in this pull request:
- Fix renderig in 1.14 of child nametags
- Don't plant wither roses
- Avoid guards getting stuck
- Fix builder placing blocks on cancel.
- Fix crafter count calc
- Fix crafter dumping and telling dman to bring it
- Fix scarecrow seed acceptance
- Fix several lang string issues.
- Fix calculation of builder blocks

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java [hunk 1]
```diff
@@ -512,7 +512,7 @@ private boolean placeBlockAt(@NotNull final BlockState blockState, @NotNull fina
     public static boolean checkForListInInvAndRequest(@NotNull final AbstractEntityAIStructure<?> placer, final List<ItemStack> itemList, final boolean force)
     {
         final List<ItemStack> foundStacks = InventoryUtils.filterItemHandler(placer.getWorker().getInventoryCitizen(),
-          itemStack -> itemList.stream().anyMatch(targetStack -> targetStack.isItemEqual(targetStack)));
+          itemStack -> itemList.stream().anyMatch(targetStack -> targetStack.isItemEqual(itemStack)));
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java:515]: nice find :D

**Review by someaddons — DISMISSED** [review]

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4763686&bid=15881782)

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4763686&bid=15881782)

**Review by marchermans — APPROVED** [review]

**Comment by marvin-bitterlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructureWithWorkOrder.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4763686&bid=15881782)

**Review by someaddons — APPROVED** [review]

**Pull request merged**